js/gateio.js

```diff
@@ -3071,6 +3072,258 @@ module.exports = class gateio extends Exchange {
+        //              "orderbook_id": 2129638396
+        //          }
+        //      ]
+        //
+        //  Delivery Futures
+        //      [
+        //          {
+        //            "name": "BTC_USDT_20200814",
+        //            "underlying": "BTC_USDT",
+        //            "cycle": "WEEKLY",
+        //            "type": "direct",
+        //            "quanto_multiplier": "0.0001",
+        //            "mark_type": "index",
+        //            "last_price": "9017",
+        //            "mark_price": "9019",
+        //            "index_price": "9005.3",
+        //            "basis_rate": "0.185095",
+        //            "basis_value": "13.7",
+        //            "basis_impact_value": "100000",
+        //            "settle_price": "0",
+        //            "settle_price_interval": 60,
+        //            "settle_price_duration": 1800,
+        //            "settle_fee_rate": "0.0015",
+        //            "expire_time": 1593763200,
+        //            "order_price_round": "0.1",
+        //            "mark_price_round": "0.1",
+        //            "leverage_min": "1",
+        //            "leverage_max": "100",
+        //            "maintenance_rate": "1000000",
+        //            "risk_limit_base": "140.726652109199",
+        //            "risk_limit_step": "1000000",
+        //            "risk_limit_max": "8000000",
+        //            "maker_fee_rate": "-0.00025",
+        //            "taker_fee_rate": "0.00075",
+        //            "ref_discount_rate": "0",
+        //            "ref_rebate_rate": "0.2",
+        //            "order_price_deviate": "0.5",
+        //            "order_size_min": 1,
+        //            "order_size_max": 1000000,
+        //            "orders_limit": 50,
+        //            "orderbook_id": 63,
+        //            "trade_id": 26,
+        //            "trade_size": 435,
+        //            "position_size": 130,
+        //            "config_change_time": 1593158867,
+        //            "in_delisting": false
+        //          }
+        //        ]
+        //
+        const tiers = this.parseLeverageTiersAllSymbols (response, 'name');
+        if (symbol !== undefined) {
+            const result = {};
+            result[symbol] = this.safeValue (tiers, symbol);
+            return result;
+        } else {
+            return tiers;
+        }
+    }
+
+    parseLeverageTiersAllSymbols (response, idKey) {
+        const tiers = {};
+        for (let i = 0; i < response.length; i++) {
+            const item = response[i];
+            const id = this.safeString (item, idKey);
+            const market = this.safeMarket (id);
+            if ((market !== undefined) && market['contract']) {
+                const symbol = market['symbol'];
+                tiers[symbol] = this.parseLeverageTiers (item, market);
+            }
+        }
+        return tiers;
+    }
```

Comment:
@frosty00 @kroitor 

How do you feel about this method? If you like it as is I can add it to base.Exchange

Reply:
@kroitor How do you feel about this parse method?

Reply:
@kroitor So if you like this parse method, I'll redo all the exchanges so that they match this

Reply:
We currently don't support transpiling `.includes()` for arrays for the same reasons as with `array.length`. We have  the  `this.inArray ()` base method for that purpose.